File: Excel/700-799/753/753 Challenge.py
import pandas as pd

path = "700-799/753/753 Pivot on Years.xlsx"

input = pd.read_excel(path, usecols="A", skiprows=1, nrows=5)
test = pd.read_excel(path, usecols="C:H", skiprows=1, nrows=5).sort_values(by="Name").reset_index(drop=True)


result = (
    input["Data"]
    .str.split(" : |, |-", expand=True)
    .rename(columns={0: "Name", 1: "Year", 2: "Value"})
    .ffill()
    .pivot_table(index="Name", columns="Year", values="Value", aggfunc="sum")
    .reset_index()
)
result.columns = test.columns

print(result.equals(test))

File: Excel/700-799/753/753 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/700-799/753/753 Pivot on Years.xlsx"
input = read_excel(path, range = "A2:A6")
test  = read_excel(path, range = "C2:H6")

result = input %>%
  separate_wider_delim(col = "Data",
                       delim = " : ", 
                       names = c("Name", "Years")) %>%
  separate_longer_delim(col = "Years",
                       delim = ", ") %>%
  separate_wider_delim(col = "Years",
                       delim = "-",
                       names = c("Year", "Value"),
                       too_few = "align_end") %>%
  fill(Year) %>%
  mutate(Value = as.numeric(Value)) %>%
  pivot_wider(names_from = Year, 
              values_from = Value, 
              values_fn = sum) %>%
  select(Name, `2021`, `2022`, `2023`, `2024`, `2025`) 

all.equal(result, test)
# > [1] TRUE